### PR TITLE
V2 overlays

### PR DESCRIPTION
**Pull request by yogevbd**

### CODE REVIEW

**Comment by yogevbd** [comment]
@DanielZlotin I added interactable just to give an example in our example project on how to implement overlays with animations (like in-app notifications)

**Comment by DanielZlotin** [comment]
due to it being unmaintained and in an alpha stage I don't think it's a good idea, unless there is a specific use case which requires us to support it explicitly

File: playground/src/screens/OptionsScreen.js [hunk 1]
```diff
@@ -190,6 +194,14 @@ class OptionsScreen extends Component {
     });
   }
 
+  onClickShowOverlay(passThroughTouches) {
+    Navigation.showOverlay({
+      component: {
+        name: 'navigation.playground.CustomDialog'
+      }
+    }, { passThroughTouches });
```
Comment (DanielZlotin) [inline playground/src/screens/OptionsScreen.js]: passThroughTouches should be part of `options` I think

Comment (yogevbd) [reply playground/src/screens/OptionsScreen.js]: done

Comment (DanielZlotin) [reply playground/src/screens/OptionsScreen.js]: and let's rename that to `interceptTouches`

**Comment by DanielZlotin** [comment]
and let's rename that to interceptTouches

File: lib/src/commands/Commands.js [hunk 2]
```diff
@@ -62,22 +62,20 @@ class Commands {
...
+  showOverlay(componentData) {
+    const input = _.cloneDeep(componentData);
+    OptionsProcessor.processOptions(input);
+
+    const layout = this.layoutTreeParser.parse(input);
+    this.layoutTreeCrawler.crawl(layout);
+
+    const promise = this.nativeCommandsSender.showOverlay(layout);
+
+    return promise;
```
Comment (anonrig) [inline lib/src/commands/Commands.js]: Why not `return this.nativeCommandsSender.showOverlay(layout)`?

**Pull request merged**